5.5/run-mysqld-slave.sh

```diff
@@ -6,6 +6,7 @@ source ${HOME}/common.sh
 set -eu
 
 export MYSQL_RUNNING_AS_SLAVE=1
+export MYSQL_MASTER_ADDR=${MYSQL_MASTER_ADDR:-mysql-master}
```

Comment:
i'm guessing the /etc/resolv.conf that gets mounted into the container includes a search domain that specifies the namespace.  it would be easy for you to confirm that.  (and easier than using the downward api).

but this still needs to be based on the service_name env variable, not hardcoded.